separated-interface/README.md

```diff
@@ -0,0 +1,119 @@
+---
+layout: pattern
+title: Separated Interface
+folder: separated-interface
+permalink: /patterns/separated-interface/
+categories: Architectural
+tags:
+ - Decoupling
+---
+
+
+## Intent
+Separate the interface definition and implementation in different packages. This allows the client to be completely unaware of the implementation.
+
+## Explanation
+
+Real world example
+
+> An Invoice generator may be created with ability to use different Tax calculators that may be added in the invoice depending upon type of purchase, region etc.         
+
+In plain words
+
+> Separated interface pattern encourages to keep the implementations of an interface decoupled from the client and its definition, so the client is not dependent on the implementation.
+
+A client code may abstract some specific functionality to an interface, and define the definition of the interface as an SPI. Another package may implement this interface definition with a concrete logic, which will be injected into the client code at runtime (with a third class, injecting the implementation in the client) or at compile time (using Plugin pattern with some configurable file).
```

Comment:
Please add explanation for term `SPI` here, since it may not be familiar to everybody.

Reply:
I think we can add https://en.wikipedia.org/wiki/Service_provider_interface link as well?

Reply:
Added the definition and link to wikipedia for better understanding